src/Agent.Plugins/TfsVCSourceProvider.cs

```diff
@@ -420,7 +420,14 @@ public sealed class TfsVCSourceProvider : ISourceProvider
                         // Cleanup the comment file.
                         if (File.Exists(commentFile))
                         {
-                            File.Delete(commentFile);
+                            try
+                            {
+                                await IOUtil.DeleteFileWithRetry(commentFile, cancellationToken);
+                            }
+                            catch (Exception ex)
+                            {
+                                executionContext.Output($"Unable to delete comment file, ex:{ex.GetType()}");
+                            }
```

Comment:
suggestion: to be cleaner, wrap this into a helper TryDeleteFileWithRetry

Reply:
the callers use tracer method with different signature.
I can wrap all of them into Action<string> parameter, but I am pretty sure it will same size of code part on caller side.